example-apps/internal-knowledge-search/app-ui/src/components/SearchApplicationSettings.tsx

```diff
@@ -1,23 +1,222 @@
-import React from 'react';
+import React, {useEffect, useState} from 'react';
 import {useDispatch, useSelector} from "react-redux";
 import {updateSettings} from "../store/slices/searchApplicationSettingsSlice";
 import {useToast} from "../contexts/ToastContext";
 import {MessageType} from "./Toast";
 import {RootState} from "../store/store";
 
+
+
 const SearchApplicationSettings: React.FC = () => {
     const dispatch = useDispatch();
-    const {appName, apiKey, searchEndpoint} = useSelector((state: RootState) => state.searchApplicationSettings);
+    const {appName, appUser, appPassword, searchEndpoint, searchPersona, searchPersonaAPIKey} = useSelector((state: RootState) => state.searchApplicationSettings);
     const {showToast} = useToast();
 
+    const fetchPersonaOptions = async () => {
+        try {
+            const identitiesIndex = await fetchIdentitiesIndex(await fetchSearchApplicationIndices())
+            const identitiesPath = searchEndpoint + "/" + identitiesIndex + "/_search"
+            const response = await fetch(identitiesPath, {
+                method: "POST",
+                headers: {
+                    "Authorization": "Basic " + btoa(appUser + ":" + appPassword),
+                    "Content-Type": "application/json"
+                },
+                body: JSON.stringify({
+                    "size": 30 //TODO: fix magic number. This is just how many identities I have in my data
+                })
+            });
+            const jsonData = await response.json();
```

Comment:
I'm happy to DRY up this pattern of

```
const path = ...
const response =  fetch(path, ...
const jsonData = await respons.json();
```
if there's a suggestion on where a "client" lib/class should live and how to expose it.